File: 804 - Unique Morse Code Words.py
class Solution:
    def uniqueMorseRepresentations(self, words: list[str]) -> int:
        code = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        s = set()
        for word in words:
            trans = ''
            for c in word:
                index = ord(c) - 97
                trans += code[index]
            print(trans)
            s.add(trans)
        return len(s)


s = Solution()
s.uniqueMorseRepresentations(["gin", "zen", "gig", "msg"])

File: 804 - Unique Morse Code Words.js
/**
 * @param {string[]} words
 * @return {number}
 */
var uniqueMorseRepresentations = function (words) {
    let code = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."];
    let s = new Set();
    for (let word of words) {
        let trans = '';
        for (let c of word) {
            trans += code[c.charCodeAt() - 97];
        }
        // console.log(trans);
        s.add(trans);
    }
    // for (let t of s) {
    //     console.log(t);
    // }
    return s.size;
};

uniqueMorseRepresentations(["gin", "zen", "gig", "msg"]);